### PR TITLE
decouple index.htm from the javascript and move away from JSXTransformer

### PR DESCRIPTION
**Pull request by chantra**
Still WIP but making a PR for visibility.

The individual commits explain the rationales. As part of this diff, I need to get the toolchain for thjs in a better state but this is a good start, that should also allow us to easily ship a atc-react module.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -55,3 +55,31 @@ After you have built ATC, you can run it by running (from within the default wor
...
+
+To build the js file:
+```
+docker-compose run node make jsx
+```
+
+If you want to have the `static/js/index.js` file automatically generated as you modify the jsx files, you can run:
+```
+docker-compose run node bash -c 'cd src/react; npm run watch'
+```
```
Comment (zfjagann) [inline HACKING.md]: Docs :+1:

File: Makefile [hunk 2]
```diff
@@ -48,6 +49,12 @@ bin/atc: src/log/*.go src/atc/*.go
 	@mkdir -p bin
 	$(BUILD) -o $@ ${SRC}/atc
 
+jsx: src/react/jsx/*.js
+	@cd src/react && $(NPM) run build-js
```
Comment (zfjagann) [inline Makefile]: I think you probably want to log the `npm` command and not the `cd`. Putting the `@` at the beginning of the line suppresses logging for the entire rest of the line. Unfortunately I don't know a way to exec `npm` in the other directory without the `cd` on the same line.

File: src/react/jsx/atc.js [hunk 3]
```diff
@@ -7,45 +7,13 @@
...
-        <div className={cls} role="alert">
-          <div className="row">
-            <div className="col-md-11">{item.message}</div>
-            <div className="col-md-1">{timeout}</div>
-          </div>
-        </div>
-      );
-    });
-  },
-});
```
Comment (zfjagann) [inline src/react/jsx/atc.js]: YES! Dead code is dead!

Comment (chantra) [inline src/react/jsx/atc.js]: thank you linter ;)

**Comment by zfjagann** [comment]
This LGTM, is it ready to land or are you still working on it?

File: Makefile (hunk 2 above)
Comment (chantra) [inline Makefile]: yeah, I dont have a way to run npm from the right directory (a la -C with tar for instance).
Each makefile line is ran in a sub-shell so we would not be in the right directory if we ran 1 command per line.
I will remove the @ though

Comment (chantra) [inline Makefile]: logging was available though, only the line showing what would be executed was missing. e.g `cd src/react && npm run build-js`

**Comment by chantra** [comment]
This is ready to land in my opinion. I will update the Makefile to remove the @ in front of the commands and land it.

File: Makefile (hunk 2 above)
Comment (zfjagann) [inline Makefile]: It's up to you whether or not to remove the `@`, I just thought logging the npm command would be helpful. I don't care so much about the `cd`. It's unfortunate there's not an easy way around this.

Comment (chantra) [inline Makefile]: so the output of the command was logged. But also logging the steps take by make is something I find usefull.

**Pull request merged**